gateway/modules/crud/bolt/update.go

```diff
@@ -0,0 +1,95 @@
+package bolt
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+
+	"github.com/boltdb/bolt"
+	"github.com/sirupsen/logrus"
+
+	"github.com/spaceuptech/space-cloud/gateway/model"
+	"github.com/spaceuptech/space-cloud/gateway/utils"
+)
+
+func (b *Bolt) Update(ctx context.Context, project, col string, req *model.UpdateRequest) (int64, error) {
+
+	switch req.Operation {
+	case utils.One, utils.Upsert:
```

Comment:
Upsert acts as regular update `all` operation when there one or more documents already exist which match the find clause